File: algorithms/c++/birthday_candles.cpp
#include <iostream>
#include<algorithm>
#include<vector>

using namespace std;

int birthdayCakeCandles(vector <int> ar) {
    // Complete this function
    int tallestCandleHeight = *max_element(ar.begin(),ar.end());
    int numOfTallestCndles = 0;
    for(vector<int>::iterator it=ar.begin();it!=ar.end();it++){
      if(*it == tallestCandleHeight )
      numOfTallestCndles++;
    }
    return numOfTallestCndles;
}

int main() {
    int n;
    cin >> n;
    vector<int> ar(n);
    for(int ar_i = 0; ar_i < n; ar_i++){
       cin >> ar[ar_i];
    }
    int result = birthdayCakeCandles(ar);
    cout << result << endl;
    return 0;
}

File: algorithms/java/src/BirthdayCandles.java
import java.io.*;
        import java.util.*;
        import java.text.*;
        import java.math.*;
        import java.util.regex.*;

public class BirthdayCandles {

    private static int birthdayCakeCandles(List<Integer> ar) {
        // Complete this function
       int tallestCandleLength  = Collections.max(ar);
       return (int) ar.stream().filter(candleLength->candleLength == tallestCandleLength).count();
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        List<Integer> ar = new ArrayList<>(n);
        for(int ar_i = 0; ar_i < n; ar_i++){
            ar.add(in.nextInt());
        }
        int result = birthdayCakeCandles(ar);
        System.out.println(result);
    }
}
